### DISCUSSION TITLE
use pyxcp to read/ set from specific address

### DISCUSSION THREAD

**Discussion by shirayr** [christoph2/pyxcp#183] [Q&A]
Hi,
i use ethernet, UDP communication, IPV6 host
receive on:
Debug PC
IPv4: 192.168.X.X
IPv6: XXXXXXXXXXXXXXXX
UDP Port: 5555
VLANID: 2
MAC Address: XXXXXXXxX
 
send to ECU:
PH XCP Slave (KVS):
IPv6: XXXXXXXXXX
UDP Port: XXXX

i need the ability to read the value on a specific address (and also to set the value) the structure defined on a2l file.
does pyxcp can help my case? do you have an example how to use pyxcp in this case?

Thanks!!

**Comment by JavierCorado** [comment]
pyXCP does not use the A2L file, you have to input the A2L values yourself, meaning that packet size calculation, data packing and unpacking, and raw to physical conversion is the user's responsibility.

Please refer to my comment in [issue 184](https://github.com/christoph2/pyxcp/issues/184) where I give a broad explanation on how to do reading and writing.
